### ISSUE TITLE
Handle uneven bin widths in spectra

### ISSUE THREAD

**Issue by rebeccafair** [pace-neutrons/Euphonic#201]
Some operations, such as broadening by convolution, are not strictly correct when applied to data with uneven bin widths (irregularly sampled data). We should ensure these are handled sensibly.

**Comment by rebeccafair** [comment]
I worry that this approach could be a lot of work for a specific case, do people generally actually care about completely correct broadening along q in a bandstructure? For publication quality plots, maybe, but in that case they may want to create their own set of Spectrum2Ds over each direction, broaden them, then stick them together to make a single Spectrum2D, as I doubt the automatically generated bandstructure would be the one that they want anyway. Horace has 'spaghetti' plots, it would be interesting to know how the broadening is handled there, whether all the q-bins are the same or not. In any case I don't think this is a high priority issue at the moment.

**Comment by ajjackson** [comment]
I don't think it would be _that_ much work, but agree that it is not really a priority. I'm happy to go ahead without it for now; if people are using these things then maybe it's worth refining.

The differences would mostly only be significant for _large_ q-broadening and I don't have much sense right now of what values are typical.
